### DISCUSSION TITLE
Email Verification Results in 404

### DISCUSSION THREAD

**Discussion by lolcatlolcat** [dani-garcia/vaultwarden#1780] [Q&A]
Trying to use the email verification feature of vaultwarden but everytime the user receives the link via email, the link just takes them to a 404 page. Nothing I can see in the logs offhand that immediately point to some issue, any ideas?

Vaultwarden is running in docker on debian buster and frontended by an Haproxy instance fwiw.

![image](https://user-images.githubusercontent.com/25574768/122217569-86a21300-ce7b-11eb-9095-edaddc313264.png)
![image](https://user-images.githubusercontent.com/25574768/122217315-4cd10c80-ce7b-11eb-98b2-ba11f84c3892.png)

**Comment by BlackDex** [comment]
You either are using a very very old version of vaultwarden (previously bitwarden_rs), or the reverse proxy is not allowing something. Could you post the Support string from the /admin/diagnostics page, and also some of the logs during the time this request was done?

**Comment by lolcatlolcat** [comment]
I updated to the latest version of `vaultwarden` this morning so it shouldn't be a version issue.

Here is the support string info:

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.21.0
* Web-vault version: v2.19.0d
* Running within Docker: true
* Uses a reverse proxy: true
* IP Header check: true (X-Forwarded-For)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.33.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*****.******-*******.***",
  "domain_origin": "*****://*****.******-*******.***",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "invitation_org_name": "Team Password Vault",
  "invitations_allowed": false,
  "ip_header": "X-Forwarded-For",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "org_attachment_limit": null,
  "org_creation_users": "*******.********@******.***,******.******@******.***,*****.*******@******.***",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_folder": "data/sends",
  "show_password_hint": true,
  "signups_allowed": true,
  "signups_domains_whitelist": "******.***",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": false,
  "smtp_from": "*********@******-*******.***",
  "smtp_from_name": "Bitwarden Password Vault",
  "smtp_host": "****.********.***",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_ssl": true,
  "smtp_timeout": 15,
  "smtp_username": "******",
  "templates_folder": "data/templates",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```

I'm trying to get some logs but `docker logs -f vaultwarden` isn't showing me anything past my successful login, Even when I get the 404, it's not being reflected in the output there. Should I look somewhere else?

**Comment by BlackDex** [comment]
Try to check your reverse proxy logs.
And verify the proxy is configured correctly: https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples
